LEAF_Request_Portal/js/formGrid.js

```diff
@@ -839,22 +840,40 @@ var LeafFormGrid = function (containerID, options) {
   }
 
   /**
-   * Set the working data set
+   * Set the working data set (uncommon)
+   * 
+   * setDataBlob() is more commonly used
+   * 
    * @params array - Expects format: [{recordID}, ...]
    * @memberOf LeafFormGrid
    */
   function setData(data) {
+    usingSetData = true;
     isDataLoaded = true;
     currentData = data;
     processedCallbackBuffer = false;
     document.querySelector(`#${prefixID}table`).setAttribute('aria-rowcount', currentData.length);
   }
 
   /**
+   * Set the working data set
+   * @params object - If setData() is not used, this expects {id: {recordID, ...}}
    * @memberOf LeafFormGrid
    */
   function setDataBlob(data) {
     dataBlob = data;
+    if(!usingSetData) {
+      if(Array.isArray(data)) {
+        currentData = data;
+      }
+      else {
+        currentData = Object.keys(data).map(key => data[key]);
```

Comment:
What happens when "data" is a string or null?

Reply:
Updated to add a console error